File: 2952-minimum-number-of-coins-to-be-added/2952-minimum-number-of-coins-to-be-added.java
class Solution {
    public int minimumAddedCoins(int[] coins, int target) {
        Arrays.sort(coins);
        int n = coins.length, pos = 0, res = 0, x = 1;
        while (x <= target)	{
        	if (pos >= n || x < coins[pos]) {
        		res++;
        		x += x;
        	} else {
        		x += coins[pos++];
        	}
        }
        return res;
    }
}

File: 2952-minimum-number-of-coins-to-be-added/2952-minimum-number-of-coins-to-be-added.py
class Solution:
    def minimumAddedCoins(self, coins: List[int], target: int) -> int:
        coins.sort()
        x, ans, i, n = 0, 0, 0, len(coins)
        while x < target:
            if i == n or coins[i] > x+1:
                # add the coin x+1
                ans += 1
                x += x + 1
            else:
                x += coins[i]
                i += 1
        return ans